CMS-Enterprise/sbom-harbor
# chore: address pedantic linter warnings in the hyper module

## Summary

This PR address the clippy warnings for the `platform::hyper` module when running clippy in pedantic mode.

### Added

- `#[must_use]` attribute to `Client` to force callers to use the constructor function.
- Missing Error docstrings
- Ignore attribute to allow the uri variable name close to the url variable name.

### Removed

Removed the module level HTTP methods that allowed making requests without a `Client` instance. They originally existed to support short lived Lambda functions, but now could cause confusion about which approach to take. If we need Lambda support in the future, we can resurrect these from source, and place them in an appropriately named submodule to communicate there intended use more clearly.

## How to test

- Pull branch
- `cargo test` - All should pass
- `cargo clippy -- -W clippy::pedantic` - You should see no warnings for the `platform::hyper` module.
